src/modes/marquee.js

```diff
@@ -0,0 +1,76 @@
+const DrawPolygon = require("./draw_polygon");
+const { geojsonTypes, updateActions, modes, events } = require("../constants");
+const doubleClickZoom = require("../lib/double_click_zoom");
+const { onMouseMove, ...RectangularDraw } = Object.assign({}, DrawPolygon);
```

Comment:
Theres some shenanigans going on up here that I think I dont understand

Reply:
@msimms1123 All DrawPolygon's properties/functions are assigned to RectangularDraw, except for onMouseMove, since if that event is included it will occasionally be possible to enter a state in which normal polygon drawing is possible.